pkg/querier/blocks_store_queryable.go

```diff
@@ -626,6 +627,12 @@ func (q *blocksStoreQuerier) fetchSeriesFromStores(
 							return validation.LimitError(fmt.Sprintf(errMaxChunksPerQueryLimit, util.LabelMatchersToString(matchers), maxChunksLimit))
 						}
 					}
+
+					for _, c := range s.Chunks{
```

Comment:
Same comment here. I would count the total size of all chunks and then call `AddChunkBytes()` once.